src/main/cpp/SwerveDrive.cpp

```diff
@@ -1,122 +1,122 @@
         m_backRight.Stop();
         m_backLeft.Stop();
         m_frontRight.Stop();
         m_frontLeft.Stop();
         return;
     }
 
-    m_backRight.drive(-backRightSpeed, backRightAngle,
-        DriveConstants::BROFF);
-    m_backLeft.drive(backLeftSpeed, backLeftAngle,
-        DriveConstants::BLOFF);
+    m_backRight.drive(-backRightSpeed, backRightAngle, DriveConstants::BROFF);
+    m_backLeft.drive(backLeftSpeed, backLeftAngle, DriveConstants::BLOFF);
     m_frontRight.drive(-frontRightSpeed, frontRightAngle,
-        DriveConstants::FROFF);
-    m_frontLeft.drive(frontLeftSpeed, frontLeftAngle, 
-        DriveConstants::FLOFF);
-}
-
-
-//Updates the Odometry
-void
-SwerveDrive::UpdateOdometry(double theta){
-    double BR_A = m_backRight.getAngle(DriveConstants::BROFF);
-    double BR_V = m_backRight.getVelocity();
-    double BL_A = m_backLeft.getAngle(DriveConstants::BLOFF);
-    double BL_V = m_backLeft.getVelocity();
-    double FR_A = m_frontRight.getAngle(DriveConstants::FROFF);
-    double FR_V = m_frontRight.getVelocity();
-    double FL_A = m_frontLeft.getAngle(DriveConstants::FLOFF);
-    double FL_V = m_frontLeft.getVelocity();
-    m_odometry.updateOdometry(BR_A, BR_V, BL_A, BL_V, FR_A, FR_V,
-        FL_A, FL_V, theta);
-}
-
-
-//Trajectory Following for Swerve
-//Cannot turn and move at the same time
-//If you want to turn make Waypoint as 0,0,0, theta
-void
-SwerveDrive::TrajectoryFollow(double rot, size_t waypointIndex){
+                       DriveConstants::FROFF);
+    m_frontLeft.drive(frontLeftSpeed, frontLeftAngle, DriveConstants::FLOFF);
+}
+
+// Updates the Odometry
+void SwerveDrive::UpdateOdometry(double theta_rad, double dt) {
+    // NOTE I am explicitly ignoring any polarity flips,
+    // because those should be fixed elsewhere, not here.
+    // The offsets are sufficient to ensure that modules
+    // return robot-relative wheel velocities as (angle, vel)
+    // pairs.
+    double BR_A = m_backRight.getAngleRad(DriveConstants::BROFF);
+    double BR_D = m_backRight.getWheelDistance();
+    double BL_A = m_backLeft.getAngleRad(DriveConstants::BLOFF);
+    double BL_D = m_backLeft.getWheelDistance();
+    double FR_A = m_frontRight.getAngleRad(DriveConstants::FROFF);
+    double FR_D = m_frontRight.getWheelDistance();
+    double FL_A = m_frontLeft.getAngleRad(DriveConstants::FLOFF);
+    double FL_D = m_frontLeft.getWheelDistance();
+    m_odometry.updateOdometry(BR_A, BR_D, BL_A, BL_D, FR_A, FR_D, FL_A, FL_D,
+                              theta_rad, dt);
+}
+
+// Trajectory Following for Swerve
+// Cannot turn and move at the same time
+// If you want to turn make Waypoint as 0,0,0, theta
+void SwerveDrive::TrajectoryFollow(double rot, size_t waypointIndex) {
     size_t index = m_trajectory_1.getIndex();
-    
+
     bool YError = (abs(GetYPosition() - m_trajectory_1.getY(index)) < 0.3);
-    bool XError = (abs(GetXPostion() - m_trajectory_1.getX(index)) < 0.3);

```

Comment:
Method rename, code wasn't compiling because it mixed `GetXPostion` and `GetXPosition`